### PR TITLE
api: port RestoreFromMnemonic to Rust

### PR DESCRIPTION
**Pull request by benma**
Builds on #683

### CODE REVIEW

File: src/rust/bitbox02-rust/src/hww/api/restore.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    let password = loop {
+        match password::enter_twice().await {
+            Err(()) => {
+                let params = confirm::Params {
+                    title: "",
+                    body: "Passwords\ndo not match.\nTry again?",
+                    ..Default::default()
+                };
+                if !confirm::confirm(&params).await {
+                    return Err(Error::Generic);
```
Comment (x1ddos) [inline src/rust/bitbox02-rust/src/hww/api/restore.rs:63]: how about `UserAbort`?

File: src/rust/bitbox02-rust/src/hww/api/restore.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        let datetime_string = bitbox02::format_datetime(timestamp, timezone_offset, false);
+        let params = confirm::Params {
+            title: "Is now?",
+            body: &datetime_string,
+            ..Default::default()
+        };
+        if !confirm::confirm(&params).await {
+            return Err(Error::Generic);
+        }
+        // Ignore error
```
Comment (x1ddos) [inline src/rust/bitbox02-rust/src/hww/api/restore.rs:77]: That is clear from the next line, that the error is ignored... but why? :)

File: src/rust/bitbox02-rust/src/hww/api/restore.rs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        let _ = bitbox02::securechip::u2f_counter_set(timestamp);
+    }
+
+    bitbox02::memory::set_initialized()?;
```
Comment (x1ddos) [inline src/rust/bitbox02-rust/src/hww/api/restore.rs:81]: Would've been nice if this resulted in `Error:Memory`, on error.

File: src/rust/bitbox02-rust/src/hww/api/restore.rs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+// limitations under the License.
+
+use super::Error;
+use crate::pb;
+
+use pb::response::Response;
+
+use crate::workflow::{confirm, mnemonic, password, status, unlock};
+
+pub async fn from_mnemonic(
```
Comment (x1ddos) [inline src/rust/bitbox02-rust/src/hww/api/restore.rs:22]: would be very very nice to have at least a couple tests

**Review by x1ddos — APPROVED** [review]
Cool.

File: src/rust/bitbox02-rust/src/hww/api/restore.rs [hunk 5]
```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 Shift Crypto AG
```
[the remark this answers is not in the document]
Comment (benma) [reply src/rust/bitbox02-rust/src/hww/api/restore.rs]: Done

File: src/rust/bitbox02-rust/src/hww/api/restore.rs (hunk 1 above)
Comment (benma) [reply src/rust/bitbox02-rust/src/hww/api/restore.rs]: Done, though technically this is a breaking change, we'll do the same as when we changed ShowMnemonic: just not bump the major version for this :)

File: src/rust/bitbox02-rust/src/hww/api/restore.rs (hunk 2 above)
Comment (benma) [reply src/rust/bitbox02-rust/src/hww/api/restore.rs]: Done

File: src/rust/bitbox02-rust/src/hww/api/restore.rs (hunk 3 above)
Comment (benma) [reply src/rust/bitbox02-rust/src/hww/api/restore.rs]: Done

**Pull request merged**